packages/toolpad-core/src/shared/navigation.tsx

```diff
@@ -8,3 +8,33 @@ export const isPageItem = (item: NavigationItem): item is NavigationPageItem =>
 export const getItemTitle = (item: NavigationPageItem | NavigationSubheaderItem) => {
   return isPageItem(item) ? item.title ?? item.segment : item.title;
 };
+
+export function hasSelectedNavigationChildren(
+  navigationItem: NavigationItem,
+  basePath: string,
+  pathname: string,
+): boolean {
+  if (isPageItem(navigationItem) && navigationItem.children) {
+    const navigationItemFullPath = `${basePath}${basePath && !navigationItem.segment ? '' : '/'}${navigationItem.segment ?? ''}`;
```

Comment:
An utility for getting this path is coming in https://github.com/mui/mui-toolpad/pull/3838/files